engines/stark/services/settings.cpp

```diff
@@ -0,0 +1,83 @@
+/* ResidualVM - A 3D game interpreter
+ *
+ * ResidualVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the AUTHORS
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "engines/stark/services/settings.h"
+#include "engines/stark/services/services.h"
+#include "engines/stark/services/archiveloader.h"
+
+#include "common/config-manager.h"
+#include "common/debug.h"
+
+#include "audio/mixer.h"
+
+namespace Stark {
+
+Settings::Settings(Audio::Mixer *mixer) :
+		_mixer(mixer) {
+	// Initialize keys
+	_boolKey[kHighModel] = "high_model";
```

Comment:
The settings name are user facing through the configuration file. I suggest using more explicit names:
* high_model => use_high_resolution_models
* specialfx => enable_special_effects
* shadow => enable_shadows
* high_fmv => play_high_resolution_videos